### ISSUE TITLE
Check add-on compatibility against Python 3

### ISSUE THREAD

**Issue by solbero** [RobLoach/lutris-kodi-addon#35]
Kodi will [drop support](https://kodi.tv/article/attention-addon-developers-migration-python-3) for Python 2 in v.19. The current release v.18 (Leia) is the last release to offer Python 2 support.

We should check the add-on's compatibility againt Python 3 to make sure we are compliant with the new version.

**Comment by RobLoach** [comment]
This is the output from `2to3`...
``` diff
--- default.py	(original)
+++ default.py	(refactored)
@@ -4,8 +4,8 @@
 import string
 import subprocess
 import sys
-import urllib
-import urlparse
+import urllib.request, urllib.parse, urllib.error
+import urllib.parse
 import xbmc
 import xbmcaddon
 import xbmcgui
@@ -15,7 +15,7 @@
 # Globals
 base_url = sys.argv[0]
 addon_handle = int(sys.argv[1])
-args = urlparse.parse_qs(sys.argv[2][1:])
+args = urllib.parse.parse_qs(sys.argv[2][1:])
 settings = xbmcaddon.Addon(id='script.lutris')
 language = settings.getLocalizedString
 
@@ -25,7 +25,7 @@
 
 # Construct a URL for the Kodi navigation
 def build_url(query):
-    return base_url + '?' + urllib.urlencode(query)
+    return base_url + '?' + urllib.parse.urlencode(query)
 
 
 # Find the path to the Lutris executable
@@ -92,8 +92,8 @@
     totalItems = len(games)
     for game in games:
         # Filter out the unprintable Unicode characters
-        name = filter(lambda x: x in string.printable, game['name'])
-        slug = filter(lambda x: x in string.printable, game['slug'])
+        name = [x for x in game['name'] if x in string.printable]
+        slug = [x for x in game['slug'] if x in string.printable]
         game_id = game['id']
         runner = game['runner'] or ''
 
@@ -138,7 +138,7 @@
     game_id = args['id'][0]
 
     # Display a notification to let the user know the game is launching.
-    game_name = filter(lambda x: x in string.printable, args['gamename'][0])
+    game_name = [x for x in args['gamename'][0] if x in string.printable]
     addon_name = settings.getAddonInfo('name')
 
     xbmcgui.Dialog().notification(addon_name, language(30002) % (game_name), settings.getAddonInfo('icon'))
```

**Comment by solbero** [comment]
These should be fine to implement and test. The Python version in `addon.xml` needs to be bumped. However, I am not well versed enough in Python to know if any more changes should be done.

**Comment by RobLoach** [comment]
When I tried the above, I ran into some Python fatal errors around urllib. Perhaps we just bump the python version for now.

**Comment by solbero** [comment]
Localization strings have been made compatible with Python 3.* with PR https://github.com/RobLoach/lutris-kodi-addon/pull/40 by @arustler.

**Comment by solbero** [comment]
I have ported the addon code to Python3 in the branch [feature/python3](https://github.com/RobLoach/lutris-kodi-addon/tree/feature/python3). I have tested it against Kodi 19 alpha and everything is working as expected. Since the Kodi devs decided to make only allow addons with a xbmc.python version of 3.0.0 in v19 I decided not making the changes Python2/3 compatible. Doing so would mean maintaining two repos one for v18 and below, and one for 19 and above.
